File: Day_7/203. Remove Linked List Elements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    let dummy = new ListNode(-1);
    dummy.next = head;

    let pre = dummy;
    let cur = head;

    while(cur){
        if(cur.val === val)
            pre.next = cur.next;
        else
            pre = cur;
        
        cur = cur.next;
    }
    return dummy.next;
};

File: Day_7/203. Remove Linked List Elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode *pre = NULL;
        ListNode *cur = head;

        while(cur != NULL){
            if(cur->val == val){
                if(pre == NULL)
                    head = cur->next;
                else
                    pre->next = cur->next;
                
                ListNode *temp = cur;
                cur = cur->next;
                delete temp;
            }
            else{
                pre = cur;
                cur = cur->next;
            }
        }
        return head;
    }
};
